asdf/fits_embed.py

```diff
@@ -87,7 +87,7 @@ def get_source(self, block):
                     return f"{FITS_SOURCE_PREFIX}{hdu.name},{hdu.ver}"
 
             msg = "FITS block seems to have been removed"
-            raise ValueError(msg)
+            raise TypeError(msg)
```

Comment:
What rule is requiring this change? This exception (and others changed in this PR) could reach and be caught be code using this library. Changing the type risks breaking their error handling code.

Reply:
Its `TRY004` (ruff, `TC004` in original rule set). See https://github.com/guilatrova/tryceratops/blob/main/docs/violations/TC004.md.

I can revert all the `TRY004` changes and ignore them. I would like to keep it on to avoid this mistake in the future.

Reply:
That rule does not appear to be correctly applied. For this instance `block` is the correct type but it's value is incorrect in that it refers to a hdu that is no longer present. Why should this be a `TypeError` and not a `ValueError`? The

Reply:
You are correct.

Unfortunately, It is a deterministic pattern checker/fixer so it just looks for an `isinstance` followed by `ValueError` and changes the error message there. Thanks for catching this. I decided to revert all these changes and suppress them instead. I think the check is fine to leave on as it will hint that one should look more closely at what one is doing.

Looking at this case, I don't think this should be an autofix.

Reply:
Thanks. That all sounds good to me.